RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc

```diff
@@ -326,152 +343,126 @@ ReducedEGProducer::ReducedEGProducer(const edm::ParameterSet& config)
   if (applyGsfElectronCalibOnData_ || applyGsfElectronCalibOnMC_) {
-    theEvent.getByToken(gsfElectronCalibEnergyT_, gsfElectronCalibEnergyHandle);
-    theEvent.getByToken(gsfElectronCalibEnergyErrT_, gsfElectronCalibEnergyErrHandle);
-    theEvent.getByToken(gsfElectronCalibEcalEnergyT_, gsfElectronCalibEcalEnergyHandle);
-    theEvent.getByToken(gsfElectronCalibEcalEnergyErrT_, gsfElectronCalibEcalEnergyErrHandle);
+    event.getByToken(gsfElectronCalibEnergyT_, gsfElectronCalibEnergyHandle);
+    event.getByToken(gsfElectronCalibEnergyErrT_, gsfElectronCalibEnergyErrHandle);
+    event.getByToken(gsfElectronCalibEcalEnergyT_, gsfElectronCalibEcalEnergyHandle);
+    event.getByToken(gsfElectronCalibEcalEnergyErrT_, gsfElectronCalibEcalEnergyErrHandle);
   }
   edm::Handle<edm::ValueMap<float>> photonCalibEnergyHandle;
   edm::Handle<edm::ValueMap<float>> photonCalibEnergyErrHandle;
   if (applyPhotonCalibOnData_ || applyPhotonCalibOnMC_) {
-    theEvent.getByToken(photonCalibEnergyT_, photonCalibEnergyHandle);
-    theEvent.getByToken(photonCalibEnergyErrT_, photonCalibEnergyErrHandle);
+    event.getByToken(photonCalibEnergyT_, photonCalibEnergyHandle);
+    event.getByToken(photonCalibEnergyErrT_, photonCalibEnergyErrHandle);
   }
 
-  auto const& caloTopology = theEventSetup.getData(caloTopology_);
+  auto const& caloTopology = eventSetup.getData(caloTopology_);
 
   //initialize output collections
-  auto photons = std::make_unique<reco::PhotonCollection>();
-  auto photonCores = std::make_unique<reco::PhotonCoreCollection>();
-  auto ootPhotons = std::make_unique<reco::PhotonCollection>();
-  auto ootPhotonCores = std::make_unique<reco::PhotonCoreCollection>();
-  auto gsfElectrons = std::make_unique<reco::GsfElectronCollection>();
-  auto gsfElectronCores = std::make_unique<reco::GsfElectronCoreCollection>();
-  auto gsfTracks = std::make_unique<reco::GsfTrackCollection>();
-  auto conversions = std::make_unique<reco::ConversionCollection>();
-  auto singleConversions = std::make_unique<reco::ConversionCollection>();
-  auto superClusters = std::make_unique<reco::SuperClusterCollection>();
-  auto ebeeClusters = std::make_unique<reco::CaloClusterCollection>();
-  auto esClusters = std::make_unique<reco::CaloClusterCollection>();
-  auto ootSuperClusters = std::make_unique<reco::SuperClusterCollection>();
-  auto ootEbeeClusters = std::make_unique<reco::CaloClusterCollection>();
-  auto ootEsClusters = std::make_unique<reco::CaloClusterCollection>();
-  auto ebRecHits = std::make_unique<EcalRecHitCollection>();
-  auto eeRecHits = std::make_unique<EcalRecHitCollection>();
-  auto esRecHits = std::make_unique<EcalRecHitCollection>();
-  auto hbheRecHits = std::make_unique<HBHERecHitCollection>();
-  auto photonPfCandMap = std::make_unique<edm::ValueMap<std::vector<reco::PFCandidateRef>>>();
-  auto gsfElectronPfCandMap = std::make_unique<edm::ValueMap<std::vector<reco::PFCandidateRef>>>();
+  reco::PhotonCollection photons{};
```

Comment:
Could you help me understand if there's a specific reason to initialize the collections with {} (value-initialization with empty initializer list) rather than the default initializer? Just curious if it makes a difference for std::vector.

Reply:
To be honest I don't see the benefit of using the initializer list instead of the default constructor. I just used curly braces so often for constructor calls in this PR that it was just a habit. I removed the curly braces here now.